apps/arweave/src/ar_pool_worker.erl

```diff
@@ -0,0 +1,90 @@
+-module(ar_pool_worker).
+
+-behaviour(gen_server).
+
+-export([start_link/0]).
+
+-export([init/1, handle_cast/2, handle_call/3, handle_info/2, terminate/2]).
+
+-include_lib("arweave/include/ar_config.hrl").
+-include_lib("arweave/include/ar_pool.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-record(state, {}).
+
+%%%===================================================================
+%%% Public interface.
+%%%===================================================================
+
+%% @doc Start the server.
+start_link() ->
+	gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%%%===================================================================
+%%% Generic server callbacks.
+%%%===================================================================
+
+init([]) ->
+	process_flag(trap_exit, true),
+	case ar_pool:is_client() of
+		true ->
+			gen_server:cast(self(), fetch_jobs);
+		false ->
+			ok
+	end,
+	{ok, #state{}}.
+
+handle_call(Request, _From, State) ->
+	?LOG_WARNING([{event, unhandled_call}, {module, ?MODULE}, {request, Request}]),
+	{reply, ok, State}.
+
+handle_cast(fetch_jobs, State) ->
+	PrevOutput = ar_pool:get_latest_output(),
+	{ok, Config} = application:get_env(arweave, config),
```

Comment:
Should we also check that mining is enabled?

Reply:
Hm, good point. I would go further and require the flag to be set at the configuration stage since it does not make sense to be a pool client and not mine and since we have a mine flag we can as well force the user to set them both to be very explicit about what's going on, what do you think?

Reply:
I like it! Prevent all non-useful configurations 💯